.github/workflows/release.yml

```diff
@@ -39,34 +27,13 @@ jobs:
           distribution: 'zulu'
           java-version: 19
 
-      - name: Prep docs
-        run: ./gradlew dokkaHtml
+      - name: Set release version
+        if: startsWith(github.ref, 'refs/tags/')
+        run: echo "VERSION_NAME=${GITHUB_REF#refs/tags/}" >> gradle.properties
```

Comment:
We had a setup like this in RxAndroid that was mostly forced on us from the RxJava Travis CI infrastructure. Personally, I was glad to see it go away when we moved to GitHub Actions and I greatly prefer the explicit version. In other projects I'm editing the README and CHANGELOG when I release to indicate the next version so I'm not sure whether there's a ton of value in automating only one location.

If you want to go down the automating route I would like to see a workflow that's manually dispatched where the version is in the input. It would automatically update the README, migrate the CHANGELOG entries from an "Unreleased" section to a versioned section, and do the `gradle.properties` update/tag/update dance.

Reply:
That makes sense.

Taking a step back, the main quality of life improvements I'd love to get from this change is not having to build the project locally (and risking using a wrong JDK or incorrect environment setup in general) for release, and not having to host signing keys on my machine - these can be achieved even without these extra workflow improvements. Given that this setup likely needs much more work to be pleasant to use, I feel like sticking with the basic requirements, and leaving the rest for later. Let me know if you have thoughts!

Reply:
I don’t like having to do 2 commits per release. Changing the property for the current release and again for the next snapshot is a bit clumsy.

I also don’t like how the tag and the property duplicate information.

I’d like to discuss our options on a VC, maybe next week?